MAINT use nanmin to replace nan by finite values in ranking of SearchCV (#24543)
scikit-learn/scikit-learn: doc/whats_new/v1.2.rst

  :func:`linear_model._sgd_fast._plain_sgd` which is used by :class:`linear_model.SGDRegressor`
  and :class:`linear_model.SGDClassifier`. The old condition did not disambiguate between
  training and validation set and had an effect of overscaling the error tolerance.
  This has been fixed in :pr:`23798` by :user:`Harsh Agrawal <Harsh14901>`.

- |Fix| :class:`model_selection.GridSearchCV` and

>   :class:`model_selection.RandomizedSearchCV` will rank as wort equally

Reviewer: "as wort equally" - what is wort? Typo?

Reviewer 2: maybe "will rank the candidates with `np.nan` scores as tied worst performers"?

>   :class:`model_selection.RandomizedSearchCV` will rank fit/scoring routines

Reviewer 2: I don't like the future. Let's be mindful and live in the present.

```suggestion
  :class:`model_selection.RandomizedSearchCV` now rank fit/scoring routines
```

>   performers the fit/scoring routines that fail and return `np.nan` scores.

Reviewer 3: I'm struggling to parse this sentence. Maybe:

> will rank candidates as equally worst when fit or scoring fails and returns `np.nan` scores.

> - |Fix| :class:`model_selection.GridSearchCV` and
>   :class:`model_selection.RandomizedSearchCV` now rank fit/scoring routines
>   that raise an exception as equal worst. Previously they were ranked as worst performers
>   in an arbitrary order.
>   :pr:`24543` by :user:`Guillaume Lemaitre <glemaitre>`.

Reviewer 4: My attempt (I think it is more about nan scores than fit/scoring routines that raise an exception). Feel free to ignore or adapt:
```suggestion
- |Fix| For :class:`model_selection.GridSearchCV` and
  :class:`model_selection.RandomizedSearchCV` ranks corresponding to nan
  scores will all be set to the maximum possible rank.
  :pr:`24543` by :user:`Guillaume Lemaitre <glemaitre>`.
```

> - |Fix| :class:`model_selection.GridSearchCV` and
>   :class:`model_selection.RandomizedSearchCV` will rank as wort equally
>   performers the fit/scoring routines that fail and return `np.nan` scores.
>   Previous failing routines were ranked arbitrarily differently without ties.
>   :pr:`24543` by :user:`Guillaume Lemaitre <glemaitre>`.

Reviewer: What do you think of this:

```suggestion
- |Fix| :class:`model_selection.GridSearchCV` and
  :class:`model_selection.RandomizedSearchCV` will rank fit/scoring routines
  that raise an exception as equal worst. Previously they were ranked as worst performers
  in an arbitrary order.
  :pr:`24543` by :user:`Guillaume Lemaitre <glemaitre>`.
```

I think the key thing to get across to people is that failing runs were always ranked as the worst results and will continue to be ranked worst. The change is that they will now be tied for last place, instead of there being an arbitrary order to them. It is tricky to pack all that into a short paragraph :-/

  nan score is correctly set to the maximum possible rank, rather than
  `np.iinfo(np.int32).min`. :pr:`24141` by :user:`Loïc Estève <lesteve>`.

- |Fix| :class:`model_selection.GridSearchCV` and

>   :class:`model_selection.RandomizedSearchCV` will rank as wort equally

Reviewer 2: ```suggestion
  :class:`model_selection.RandomizedSearchCV` now rank as wort equally
```

> - |Fix| :class:`model_selection.GridSearchCV` and
>   :class:`model_selection.RandomizedSearchCV` now rank as wort equally
>   performers the fit/scoring routines that fail and return `np.nan` scores.
>   Previous failing routines were ranked arbitrarily differently without ties.
>   :pr:`24543` by :user:`Guillaume Lemaitre <glemaitre>`.

Reviewer 4: ```suggestion
- |Fix| For :class:`model_selection.GridSearchCV` and
  :class:`model_selection.RandomizedSearchCV` ranks corresponding to nan
  scores will all be set to the maximum possible rank.
  :pr:`24543` by :user:`Guillaume Lemaitre <glemaitre>`.
```
